merlion/models/anomaly/dagmm.py

```diff
@@ -203,6 +206,46 @@ def train(
         )
         return train_scores
 
+    def train_multiple(
+        self, train_data: List[TimeSeries], anomaly_labels: List[TimeSeries] = None,
+        train_config=None, post_rule_train_config=None
+    ) -> List[TimeSeries]:
+        """
+        Trains the anomaly detector (unsupervised) and its post-rule
+        (supervised, if labels are given) on the input multiple time series.
+
+        :param train_data: a list of `TimeSeries` of metric values to train the model.
+        :param anomaly_labels: a list of `TimeSeries` indicating which timestamps are
+            anomalous. Optional.
+        :param train_config: Additional training config dict:
+            "n_epochs": int indicating how many times the model must be trained on the `train_data`
+            "shuffle": bool indicating if the `train_data` should be shuffled before every epoch
+        :param post_rule_train_config: The config to use for training the
+            model's post-rule. The model's default post-rule train config is
+            used if none is supplied here.
+
+        :return: A list of `TimeSeries` of the model's anomaly scores on the training
+            data with each element corresponds to time series from `train_data`.
+        """
+        if anomaly_labels is not None:
+            assert len(train_data) == len(anomaly_labels)
+        else:
+            anomaly_labels = [None] * len(train_data)
+        n_epochs = train_config.pop("n_epochs", 1)
+        shuffle = train_config.pop("shuffle", False)
```

Comment:
Maybe you can make this `shuffle = train_config.pop("shuffle", n_epochs > 1)`?

Reply:
Makes sense. I will also add that shuffling is turned on by default for `n_epochs` > 1 in the docstring.